posthog/queries/sessions/session_recording.py

```diff
@@ -59,10 +61,54 @@ def query_recording_snapshots(
 
         return events[0].distinct_id, events[0].timestamp, [e.snapshot_data for e in events]
 
+    def merge_snapshot_chunks(self, team: Team, session_id: str, unmerged_snapshots: List[Dict[str, any]]):
+        snapshot_collectors: Dict[str, Dict[str, any]] = {}  # gather the chunks of a snapshot
+        snapshots_and_collectors: List[Dict[str, any]] = []  # list the snapshots in order
+
+        for chunk_or_snapshot in unmerged_snapshots:
+            if chunk_or_snapshot.get("posthog_chunked"):  # it's a chunk
+                collector = snapshot_collectors.get(chunk_or_snapshot["snapshot_id"])
+                if not collector:
+                    collector = {
+                        "collector": True,
+                        "event": chunk_or_snapshot,
+                        "count": chunk_or_snapshot["chunk_count"],
+                        "chunks": {},
+                    }
+                    snapshot_collectors[chunk_or_snapshot["snapshot_id"]] = collector
+                    snapshots_and_collectors.append(collector)
+
+                collector["chunks"][chunk_or_snapshot["chunk_index"]] = chunk_or_snapshot["chunk_data"]
+            else:  # full snapshot
+                snapshots_and_collectors.append({"snapshot": True, "data": chunk_or_snapshot})
+
+        snapshots = []
+        for snapshot_or_collector in snapshots_and_collectors:
+            if snapshot_or_collector.get("collector"):
+                has_all_chunks = True
+                data = ""
+                for i in range(snapshot_or_collector["count"]):
+                    if not snapshot_or_collector["chunks"].get(i):
+                        has_all_chunks = False
+                        break
+                    data = data + snapshot_or_collector["chunks"][i]
+
+                if has_all_chunks:
+                    snapshots.append(json.loads(data))
+                else:
+                    capture_message(
+                        "Did not find all session recording chunks! Team: {}, Session: {}".format(team.pk, session_id)
+                    )
+            elif snapshot_or_collector.get("snapshot"):
+                snapshots.append(snapshot_or_collector["data"])
+
+        return snapshots
+
     def run(self, team: Team, session_recording_id: str, *args, **kwargs) -> Dict[str, Any]:
         from posthog.api.person import PersonSerializer
 
-        distinct_id, start_time, snapshots = self.query_recording_snapshots(team, session_recording_id)
+        distinct_id, start_time, unmerged_snapshots = self.query_recording_snapshots(team, session_recording_id)
```

Comment:
Will SESSIONS_IN_RANGE_QUERY above still continue working with this setup? From a cursory glance at the code it seems not.

Basically there will always be sessions with incomplete data - e.g. person leaves the page before posthog.js finishes sending the full payload event or the larger payload gets blocked. We should not show sessions where there's no full snapshot event which the below line accomplishes:

`COUNT(*) FILTER(where snapshot_data->>'type' = '2') as full_snapshots`